internal/pkg/grpc/gen/gen.go

```diff
@@ -33,18 +33,22 @@ func NewGenerator(data *userdata.UserData, port int) (g *Generator, err error) {
 		return nil, err
 	}
 
-	// TODO: In the case of failure, attempt to generate the identity from
-	// another RoT.
+	// Loop through trustd endpoints and attempt to download PKI
 	var conn *grpc.ClientConn
-	conn, err = basic.NewConnection(data.Services.Trustd.Endpoints[0], port, creds)
-	if err != nil {
-		return nil, err
+	for i := 0; i < len(data.Services.Trustd.Endpoints); i++ {
+		conn, err = basic.NewConnection(data.Services.Trustd.Endpoints[i], port, creds)
+		if err != nil {
+			//Unable to connect, bail and attempt to contact next endpoint
+			continue
+		}
+		client := proto.NewTrustdClient(conn)
+		return &Generator{client: client}, nil
 	}
-	client := proto.NewTrustdClient(conn)
 
-	return &Generator{
-		client: client,
-	}, nil
+	// We were unable to connect to any trustd endpoint
+	// Return error from last attempt.
+	return nil, err
```

Comment:
does it make sense to combine all the errors into `multierror`, as the error talking to the last endpoint might be not the same as the others, potentially hiding the issue? (e.g. last endpoint is down, but others have other issues, and the fact that it's down gives zero info on the previous failures)

Reply:
Good thought, let me look at how we've used that package previously.

Reply:
@smira updated. see what you think now.